Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -856,7 +856,7 @@ Bytecode::CodeGenerationErrorOr<void> ArrayExpression::generate_bytecode(Bytecod
                 };
             }
         } else {
-            generator.emit<Bytecode::Op::LoadImmediate>(Value {});
+            generator.emit<Bytecode::Op::LoadImmediate>(js_undefined());
```

Comment:
This is not correct, there is an observable difference between an array hole and `undefined`.
Small script which will probably break with this:
```js

const a = [undefined, /* hole */, true];
a.length === 3;
a.hasOwnProperty(0) === true;
a.hasOwnProperty(1) === false;
```

(Whoops I see Andreas already mentioned it, hopefully the example is useful at least)

Reply:
Thanks for the example :^)